core/src/main/scala/com/mesosphere/usi/core/scalaapi/SchedulerAdapter.scala

```diff
@@ -0,0 +1,170 @@
+            }
+        }
+
+      Source.single(specsSnapshot -> events)
+    }.via(killSwitch.flow)
+      .via(schedulerFlow)
+      .flatMapConcat {
+        case (snapshot, updates) =>
+          stateSnapshotPromise.trySuccess(snapshot)
+          updates.watchTermination() {
+            case (_, cancellationSignal) =>
+              cancellationSignal.onComplete {
+                case Success(_) =>
+                  killSwitch.shutdown()
+                case Failure(cause) =>
+                  killSwitch.abort(cause)
+              }
+          }
+      }
+      .via(killSwitch.flow)
+      .mapAsync(1)(stateQueue.offer)
+      .map {
+        case QueueOfferResult.Enqueued =>
+        case QueueOfferResult.QueueClosed =>
+          killSwitch.shutdown()
+        case QueueOfferResult.Failure(cause) =>
+          killSwitch.abort(cause)
+        case QueueOfferResult.Dropped => // we shouldn't receive that at all because OverflowStrategy.backpressure
+          throw new RuntimeException("Unexpected QueueOfferResult.Dropped element")
+      }
+      .runWith(Sink.ignore)
+
+    val sourceWithKillSwitch = stateSource
+      .watchTermination() {
+        case (materializedValue, cancellationSignal) =>
+          cancellationSignal.onComplete {
+            case Success(_) =>
+              killSwitch.shutdown()
+            case Failure(cause) =>
+              killSwitch.abort(cause)
+          }
+          materializedValue
+      }
+      .via(killSwitch.flow)
+
+    val sinkWithKillSwitch = Flow[SpecUpdated]
+      .watchTermination() {
+        case (materializedValue, cancellationSignal) =>
+          cancellationSignal.onComplete {
+            case Success(_) =>
+              killSwitch.shutdown()
+            case Failure(cause) =>
+              killSwitch.abort(cause)
+          }
+          materializedValue
+      }
+      .via(killSwitch.flow)
+      .to(specSink)
+
+    (stateSnapshotPromise.future, sourceWithKillSwitch, sinkWithKillSwitch)
+
+  }
+
+  /**
+    * Represents the scheduler as a Flow.
+    *
+    * This method will materialize the scheduler first, then Flow can be materialized independently.
+    * @param specsSnapshot Snapshot of the current specs
+    * @return
+    */
+  def asFlow(specsSnapshot: SpecsSnapshot = SpecsSnapshot.empty)
+    : (Future[StateSnapshot], Flow[SpecUpdated, StateEvent, NotUsed]) = {
+    val (snapshot, source, sink) = asSourceAndSink(specsSnapshot)
+    snapshot -> Flow.fromSinkAndSourceCoupled(sink, source)
+  }
+
+  /**
+    * Represents the scheduler as a SourceQueue and a SinkQueue.
+    *
+    * @see https://doc.akka.io/api/akka/current/akka/stream/scaladsl/SourceQueue.html
+    *      https://doc.akka.io/api/akka/current/akka/stream/scaladsl/SinkQueue.html
+    *
+    * This method will materialize the scheduler first, then queues can be used independently.
+    * @param specsSnapshot Snapshot of the current specs
+    * @return
+    */
+  def asAkkaQueues(
```

Comment:
could we also document what would be the cases when one should choose this one and not the other? I think it would be nice for people integrating with USI

Reply:
I think the best way to document this would be to have an example of usage.